// @material/web/tabs/harness.js
/**
 * @license
 * Copyright 2023 Google LLC
 * SPDX-License-Identifier: Apache-2.0
 */
import { Harness } from '../testing/harness.js';
/**
 * Test harness for Tab.
 */
export class TabHarness extends Harness {
    async getInteractiveElement() {
        await this.element.updateComplete;
        return this.element.renderRoot
            .querySelector('.button');
    }
    async completeIndicatorAnimation() {
        await this.element.updateComplete;
        const animations = this.element.indicator.getAnimations();
        for (const animation of animations) {
            animation.finish();
        }
    }
    async isIndicatorShowing() {
        await this.completeIndicatorAnimation();
        const opacity = getComputedStyle(this.element.indicator)['opacity'];
        return opacity === '1';
    }
}
/**
 * Test harness for Tabs.
 */
export class TabsHarness extends Harness {
    // Note, Tabs interactive element is the interactive element of the
    // selected tab.
    async getInteractiveElement() {
        await this.element.updateComplete;
        const selectedItemHarness = this.element.selectedItem.harness ??
            new TabHarness(this.element.selectedItem);
        return await selectedItemHarness.getInteractiveElement();
    }
    get harnessedItems() {
        // Test access to protected property
        // tslint:disable-next-line:no-dict-access-on-struct-type
        return this.element['items']
            .map(item => {
            return (item.harness ?? new TabHarness(item));
        });
    }
}
